L62-core-call-credential-security-level.md

```diff
@@ -0,0 +1,121 @@
+gRPC security level negotiation between call credentials and channels.
+----
+* Author(s): Yihua Zhang
+* Approver: Mark Roth
+* Status: In review
+* Implemented in: https: //github.com/grpc/grpc/pull/21215
+* Last updated: 2019-12-05
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/ZtXYweohOwE
+
+## Abstract
+
+Associate a security level (privacy+integrity, integrity-only, insecure)
+to each `grpc_call_credentials` implementation to dictate the level of
+transport security an underlying connection should satisfy in order to transfer
+a corresponding call credential. Whenever a call credential is about to be
+transferred on a channel, we check if the security level of channel is greater than or
+equal to that of call credential and fail the transfer if it is not.
+
+## Background
+
+In gRPC Java and Go, each call credential can dictate whether or not
+a secure channel (i.e., ensuring both privacy and integrity) is needed
+for its communication. The use of secure channel is mandatory if a call
+credential is used to access Google APIs. In gRPC C core and its wrapped
+languages, a call credential can only be communicated over a secure channel
+which is also acceptable from the security perspective. However with the
+introduction of local credential that can be used in both UDS and TCP loopback
+connections, any call credential (including the ones used to access Google APIs)
+can be transferred over a TCP loopback connection that is considered insecure, which
+greatly lowers the security bar. In more detals, UDS connections can be treated
+as relatively secure (assuming a correct directory permission is set on a file path)
+as a remote peer can be authenticated and data confidentiality can be guaranteed
+against passive adversaries. However for local TCP, we cannot authenticate a
+remote peer and thus both data confidentiality and integrity become meaningless.
+
+### Related Proposals:
+
+N/A
+
+## Proposal
+
+- A call credential should have an ability to dictate the security level of
```

Comment:
s/the security level/the minimum security level/

Reply:
Done.